metrics.md

```diff
@@ -0,0 +1,238 @@
+
+    Of all the people who used Sync in the last (28) days, what percentage of them used it on a given day?
+
+Having a high ratio of daily users in relation to monthly users  indicates that a majority of users are using Sync frequently. Tracking the engagement ratio allows us to monitor how changes to the user experience affect how frequently users engage.
+
+Since the engagement ratio is a primary measure for our success it can be useful to compare ratios of different subsets of the user base. Segmenting by count of connected device and storage size are two examples. By identifying segments with higher engagement, we can focus efforts into making it easy for users to become members of those segments.
+
+*TODO* add graph screenshot
+
+### Retention Rate
+
+    How loyal are our users?
+
+Knowing how long people continue using Sync is our second primary measure. By improving retention we know that we're maintaining users' interest.
+
+Short term retention can help us understand:
+
+* the quality of our user acquisition strategy
+* the quality of our initial user experience
+* how well we inform users about what Sync does
+
+Long term retention can help us understand:
+
+* what features lead to long term use
+* long term satisfaction of individual features
+
+Retention timeframes are TBD.
+
+### Service Health
+
+High level service health helps us correlate service degradation or disruption with other metrics. Lower level stats for debugging problems are handled by other systems.
+
+*TODO* add graph screenshot
+
+## Data Collection
+
+Logs from the servers are loaded into Redshift for analysis. Each day a new log table is created. These tables are dropped after approximately 90 days.
+
+Here's the schema for the log tables:
+
+```sql
+create table sync0715
+(
+  uid CHAR(32) NOT NULL encode lzo,    -- a sha256-hashed Firefox Account (FxA) user id
+  s_uid CHAR(32) encode lzo,           -- a surrogate user id (generated by token server)
+  dev CHAR(32) NOT NULL encode lzo,    -- a sha256-hashed FxA device id
+  s_dev CHAR(32) encode lzo,           -- a surrogate device id (generated by token server)
+  ts TIMESTAMP NOT NULL encode lzo,    -- timestamp of the request
+  method VARCHAR(32) encode lzo,       -- request method (GET, POST, etc)
+  code SMALLINT encode lzo,            -- http status code of the response
+  bucket VARCHAR(255) encode bytedict, -- sync bucket name (bookmarks, history, etc)
+  t INTEGER encode bytedict,           -- request time in milliseconds
+  ua_browser VARCHAR(255) encode lzo,  -- request User Agent browser
+  ua_version INTEGER encode lzo,       -- request User Agent browser version
+  ua_os VARCHAR(255) encode lzo,       -- request User Agent OS
+  host VARCHAR(255) encode lzo         -- server hostname that handled the request
+)
+distkey(uid)
+interleaved sortkey(uid, dev);
+```
+
+## Rollups
+
+After a daily table has been loaded, a few sql commands do rollups of the data.
+
+### Device Activity
```

Comment:
Perhaps it's just me but I feel like it would help to see definitions of what each column in the table will look like. The queries (although very nice to have and more detailed than many teams) provide at times too much detail in order to quickly understand the available data.

Here are examples of how other teams have defined what data is available in there tables:
- Android: https://wiki.mozilla.org/Mobile/Metrics/Redash
- FxA funnel data: https://github.com/mozilla/fxa-auth-server/blob/7ef8d5d1c9e02f9bc119adf4b6356eb5716beb7d/docs/metrics-events.md